### PR TITLE
improve netcdf multi-group reading and handle metadata QC issue

### PR DESCRIPTION
**Pull request by JessicaS11**
Per [this discussion](https://forum.earthdata.nasa.gov/viewtopic.php?t=5154), ATL11 v006, ATL14 v003, and ATL15 v003 have metadata issues that cause icepyx to fail when it tries to get the version number from the granule file. This PR issues a temporary fix that instead gets the most recent version number from CMR to use instead.

It also adds handling reading in multiple groups of a netcdf.

### CODE REVIEW

**Comment by JessicaS11** [comment]
@rwegener2 @betolink @andypbarrett @scottyhq @wsauthoff @tsutterley 
Before reporting to xarray, I'm hoping one of you can check I'm not doing something silly. Per [this toy example](https://github.com/icesat2py/icepyx/blob/metadata/doc/source/example_notebooks/ATL15-cloudvslocal.ipynb) (also in the metadata branch that this PR is for, but easier to view not as a GitHub diff), when I read an ATL15 file in using Xarray (`xr_open_dataset()`) in the cloud versus locally, the resulting structure of the dataset is different. Namely, in the cloud the variables include the time dimension, and locally they do not. I initially noted it via the icepyx.Read functionality (bottom half of the notebook), but reproduced it with xarray directly (top half of the notebook). I've tried controlling for versions and defaults, but not yet with a fine tooth comb. I know suspect the number of x and y values differ due to the granule being ordered with subsetting (I thought I set it to false, but...), but that shouldn't make a difference anyway. Have any of you seen this?

**Comment by tsutterley** [comment]
hey @JessicaS11, is `ATL15_GL_0318_01km_003_01_HEGOUT.nc` a renamed version of the ATL15 granule?

**Comment by JessicaS11** [comment]
> hey @JessicaS11, is `ATL15_GL_0318_01km_003_01_HEGOUT.nc` a renamed version of the ATL15 granule? Or is a subset file from the NSIDC API? I _think_ NSIDC is still working on 3D raster subsetting capability.

That's what came back when I ordered and downloaded from NSIDC:
```
region = ipx.Query(
         product="ATL15",
         spatial_extent=[-17.25, 80.7, -16.0, 81.0],  # minlon, minlat, maxlon, maxlat
         date_range=['2018-09-15', '2023-03-02']
     )

     region.download_granules(path="./doc/source/example_notebooks/atl15",
                              subset=False)
```

> I _think_ NSIDC is still working on 3D raster subsetting capability.

That's also what I thought... my naive (and "potentially an issue" assumption) was that even if the file were being subset spatially, it shouldn't be affecting the structure and thus how xarray reads in a specific group.

**Comment by betolink** [comment]
just to confirm, @JessicaS11 the local granule was downloaded using icepyx? 

if it was downloaded using the on-prem subsetting service maybe there is a bug in EGI (or an undocumented behavior) that removed the time dimension even if we are not subsetting the granule.

**Comment by JessicaS11** [comment]
> just to confirm, @JessicaS11 the local granule was downloaded using icepyx?

Yes - with subsetting applied. (So subsetting on rasters is available now, @tsutterley!)

Submitting to EGI without subsetting params returned ftp downloadUrls (which icepyx isn't set up to handle, since it looks for the order ID). To initiate the unprocessed granule download (via a url request in the browser), I had to set `agent=NO` and `mode=sync`, at which point I assume it's not going through EGI. Opening the full granule with xarray behaved just as on the cloud, with the variables correctly formatted with x, y and time dimensions.
 
> if it was downloaded using the on-prem subsetting service maybe there is a bug in EGI (or an undocumented behavior) that removed the time dimension even if we are not subsetting the granule.

@mikala-nsidc, @betolink suggested you might know what's going on (or someone who does) such that somehow EGI is restructuring the data when it's subsetted to turn the time dimension into bands (as interpreted by xarray).

**Comment by mikala-nsidc** [comment]
The way that the on prem subsetter handles ATL15 parameters that are 3 dimensional arrays is by "flattening" those bands.  This means, for instance, that if you select a spatial subset of an ATL15 file, the time dimension in dh/dt will be separated into 18 bands.  Attaching a screenshot to illustrate.  
Screenshot 2024-03-07 at 10 25 34 AM
Spatial subsetting will be handled differently once those services are available in the cloud.

**Comment by JessicaS11** [comment]
@wsauthoff @mikala-nsidc Are either of you available to review this PR?

File: icepyx/core/read.py [hunk 1]
```diff
@@ -330,6 +330,17 @@ def vars(self):
         <icepyx.core.variables.Variables at [location]>
         """
 
+        # fix to handle fact that some VersionID metadata is wrong
+        # see: https://forum.earthdata.nasa.gov/viewtopic.php?t=5154
+        # (v006, v003, v003, respectively)
+        # Note that this results in a login being required even for a local file
+        # because otherwise Variables tries to get the version from the file (ln99).
```
[the remark this answers is not in the document]
Comment (JessicaS11) [reply icepyx/core/read.py:337]: Lines 99-103 of the Variables module are:
```
            # Check for valid version string
            # If version is not specified by the user assume the most recent version
            self._version = val.prod_version(
                is2ref.latest_version(self._product), version
            )
```

I'll update the comment to make it clearer that it's a different file.

Comment (JessicaS11) [inline icepyx/core/read.py:337]:
```suggestion
        # because otherwise Variables (variables.py) tries to get the version from the file (ln99).
```

**Pull request merged**